src/core/ai/agent/SaikiAgent.ts

```diff
@@ -834,6 +834,16 @@ export class SaikiAgent {
             : this.stateManager.getRuntimeConfig();
     }
 
+    /**
+     * Expose the ToolConfirmationProvider so that application layers can listen
+     * to confirmation events or forward user responses without having to dig
+     * through internal classes.
+     */
+    public getToolConfirmationProvider() {
```

Comment:
@coderabbitai this wouldn't be needed any more with event driven architecture right?